skfda/exploratory/stats/_stats.py

```diff
@@ -91,3 +91,38 @@ def depth_based_median(fdatagrid, depth_method=modified_band_depth):
 
     # The median is the deepest curve
     return fdatagrid[indices_descending_depth[0]]
+
+
+def trim_mean(fdatagrid,
+              proportiontocut,
+              depth_method=modified_band_depth):
+    """Compute the trimmed means based on a depth measure.
+
+    The trimmed means consists in computing the mean function without a
+    percentage of least deep curves. That is, we first remove the least deep
+    curves and then we compute the mean as usual.
+
+    Args:
+        fdatagrid (FDataGrid): Object containing different samples of a
+            functional variable.
+        proportiontocut (float): indicates the percentage of functions to
+            remove. It is not easy to determine as it varies from dataset to
+            dataset.
+        depth_method (:ref:`depth measure <depth-measures>`, optional):
+            Method used to order the data. Defaults to :func:`modified
+            band depth <fda.depth_measures.modified_band_depth>`.
+
+    Returns:
+        FDataGrid: object containing the computed trimmed mean.
+
+    """
+    n_samples_to_keep = int((fdatagrid.n_samples -
```

Comment:
You can use `scipy.stats.trim1` to simplify the code even more and make its behaviour as similar as possible with scipy.

Reply:
I tried this but FDataGrid is not compatible with the internal operations of trim1. Maybe just use what is implemented as it is trivial?

Reply:
Ok, but make sure that the rounding is the same.

Reply:
Done, i am using the exact same operation now to find the index where we  cut.